File: NotificationActivity/app/src/main/java/com/calvert/notificationactivity/SecondActivity.java
package com.calvert.notificationactivity;

import android.app.NotificationManager;
import android.content.Context;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Toast;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        NotificationManager notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
        notificationManager.cancel(getIntent().getIntExtra("key", -1));
        Toast.makeText(this, "Second", Toast.LENGTH_SHORT).show();
    }
}

File: FirstKotlinApp/app/src/main/java/com/calvert/firstkotlinapp/SecondActivity.kt
package com.calvert.firstkotlinapp

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast

class SecondActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        var receivedName = intent.getStringExtra("name")
        Toast.makeText(this, receivedName, Toast.LENGTH_LONG).show()
    }
}
